### PR TITLE
FE#21: Added base dashboard.

### PR DESCRIPTION
**Pull request by logicplace**
Added basic dashboard with selectable control icons and a basic menu. Menu allows user to set their default view and logout (redirects to login page). Default view is loaded and selected on startup.

### CODE REVIEW

File: dashboard.html [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <title>My Vineyard - Plantalytics</title>
+        
+        <!-- Includes -->
+        <link href="https://fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet"> 
+        <script src="http://code.jquery.com/jquery-2.0.0.js"></script>
+        <link rel="stylesheet" href="css/dashboard.css"/>
+        <script type="text/javascript" src="js/dashboard.js"></script>
+        <script src="https://maps.googleapis.com/maps/api/js?callback=readyMap" async defer></script>
```
Comment (kbrimm) [inline dashboard.html]: Will change to Leaflet - can do this as part of the Map API task.

File: dashboard.html [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                        </div>
+                    </div>
+                    <div class="menu-group">
+                        <div class="menu-entry" id="menu-logout">Logout</div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </body>
+</html>
```
Comment (kbrimm) [inline dashboard.html]: EOF newline

File: js/dashboard.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            // Set new default.
+            localStorage.defaultDataView = selector;
+        }
+    });
+
+    $("#menu-logout").click(function() {
+        // TODO: log user out.
+        window.location = "login.html";
+    });
+});
```
Comment (kbrimm) [inline js/dashboard.js]: EOF newline

File: css/dashboard.css [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    padding: 0px 5px;
+}
+
+.menu-entry:hover {
+    background-color: lightgray;
+}
+
+.centered {
+    text-align: center;
+}
```
Comment (kbrimm) [inline css/dashboard.css]: EOF newline

File: css/dashboard.css [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    display: inline-block;
+}
+
+.data-button-label > input {
+    visibility: hidden;
+    position: absolute;
+}
+
+.data-button-label > input:checked + .data-button {
+    background-color: lightgray;
```
Comment (kbrimm) [inline css/dashboard.css]: This color change is a bit subtle - maybe change to something a little more obvious?

**Comment by kbrimm** [comment]
This looks great! I do have a couple of thoughts about improvement:
- I like the overall styling of the settings menu, but it feels a bit cramped
  - The whitespace around Logout is inconsistent with the rest of the menu. Adding some padding to that element would make the flow feel more unified.
  - The icons feel smooshed. Would it be possible to resize the images in the menu?
- Eventually we should think about unifying the theming between the login and dashboard pages, but that's a bridge we can cross later on.
- The menu/buttons don't display in IE 11 on Windows 7. I know we axed compatibility concerns with IE, it just makes me nervous and I thought I should point it out.

File: css/dashboard.css (hunk 5 above)
Comment (logicplace) [inline css/dashboard.css]: Suggestions?

**Comment by logicplace** [comment]
Unsurprisingly IE11 doesn't support var or calc. :( I'll remove the vars I guess. Also yeah this round wasn't so much about theming, more just placement & function. I think that will be a separate ticket.

**Comment by logicplace** [comment]
I think I addressed everything. I tested on IE11 to make sure it works, too (other than saving the preference). I also added a temporary spacer to the menu so hopefully it looks a little less cramped overall?

**Comment by kbrimm** [comment]
Looks good!

**Pull request merged**